Patch for adding `-c` to `ranlib` flags must not be applied when using `llvm-ranlib`

Trying to build `develop` or release `11.5.0` on macOS with LLVM clang + gfortran:
```
     846    /usr/local/opt/llvm/bin/llvm-ranlib -c libbufr_4_DA.a
  >> 847    /usr/local/opt/llvm/bin/llvm-ranlib: error: Invalid option: '-c'
  >> 848    make[2]: *** [src/libbufr_4_DA.a] Error 1
     849    make[2]: *** Deleting file `src/libbufr_4_DA.a'
  >> 850    make[1]: *** [src/CMakeFiles/bufr_4_DA.dir/all] Error 2
  >> 851    make: *** [all] Error 2
```
In CMakeLists.txt:
```
if(APPLE)
  # The linker on macOS does not include `common symbols` by default
  # Passing the -c flag includes them and fixes an error with undefined symbols
  set(CMAKE_Fortran_ARCHIVE_FINISH "<CMAKE_RANLIB> -c <TARGET>")
  set(CMAKE_C_ARCHIVE_FINISH "<CMAKE_RANLIB> -c <TARGET>")
endif()
```
I tested this with different `clang` versions. The native Apple clang requires the `-c` flag, otherwise the linker step fails with undefined symbols. The LLVM clang ranlib (`llvm-ranlib`) doesn't understand the `-c` flag and doesn't need it.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,7 @@ if(ENABLE_PYTHON)
   find_package(Python3 REQUIRED COMPONENTS Interpreter)
 endif()
 
-if(APPLE)
+if(APPLE AND NOT "${CMAKE_RANLIB}" MATCHES "^.*(llvm-ranlib)$")
   # The linker on macOS does not include `common symbols` by default
   # Passing the -c flag includes them and fixes an error with undefined symbols
   set(CMAKE_Fortran_ARCHIVE_FINISH "<CMAKE_RANLIB> -c <TARGET>")

